battenberg_gitlab/apply.py

```diff
@@ -0,0 +1,101 @@
+import os
+import logging
+import json
+from typing import List
+from functools import reduce
+from gitlab import Gitlab
+from pygit2 import Repository
+from battenberg import Battenberg
+from battenberg.errors import MergeConflictException
+from battenberg_gitlab.search import get_projects
+from battenberg_gitlab.utils import clone_or_discover_repo, ensure_workspace
+
+
+logger = logging.getLogger(__name__)
+
+
+# TODO Remove these once we've we've backfilled .cookiecutter.json
+with open(os.path.join(os.path.dirname(__file__), '..', 'templates.json')) as f:
+    templates = json.load(f)
+
+def construct_template(project_name: str) -> dict:
+    template = {
+        **templates['defaults'],
+        'project_name': project_name,
+        'module_name': project_name.replace('-', '_'),
+        **templates['projects'][project_name]
```

Comment:
There's a structure assumed by this function. It might be helpful to document those assumptions or provide a sample template that follows the convention this function expects.

Maybe this isn't that big of an issue, since it goes away once you have .cookiecutter.json populated.

Reply:
Yeah this is hacky code really just because the v1.0 PRs didn't have the logic for generating `.cookiecutter.json` which was a mistake not to port in (caused a lot more work on my plate!).

My intention is to remove this code before merging this PR.